provider/virtual_network/network_base.py

```
@@ -362,6 +363,53 @@ def check_iface_attrs(iface, key, expect_val):
+    """
+    if iface_type == "bridge":
+        bridge_name = params.get('bridge_name', 'br0')
+        host_iface = params.get('host_iface')
+        utils_net.delete_linux_bridge_tmux(bridge_name, host_iface)
+
+    elif iface_type == "ethernet":
+        tap_name = params.get('tap_name', 'mytap0')
+        delete_tap(tap_name)
+
+    elif iface_type == "network":
+        network_dict = eval(params.get("network_dict", "[]"))
```

Ruff (0.13.1)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)